imports/plugins/core/product/server/no-meteor/resolvers/Mutation/createProductVariant.js

```diff
@@ -0,0 +1,29 @@
+import { decodeProductOpaqueId } from "@reactioncommerce/reaction-graphql-xforms/product";
+
+/**
+ *
+ * @method createProductVariant
+ * @summary initializes empty variant template
+ * @param {Object} _ - unused
+ * @param {Object} args - The input arguments
+ * @param {Object} args.input - mutation input object
+ * @param {String} args.input.clientMutationId - The mutation id
+ * @param {String[]} args.input.parentId - the product or variant ID which we create new variant on
+ * @param {Object} context - an object containing the per-request state
+ * @return {Promise<Object>} createProductVariant payload
+ */
+export default async function createProductVariant(_, { input }, context) {
+  const {
+    clientMutationId = null,
+    parentId
```

Comment:
@aldeed should we pass `shopId` here as well? I know we usually want to pass shopId, however shopId is available on the `parent` product, which is found prior to anywhere that would use it: https://github.com/reactioncommerce/reaction/pull/5562/files#diff-23f5be7d77820413c96a2327c790435bR27

Reply:
We've been trying to always ask for `shopId` even if we can get it because in many cases it allows the auth check to happen before any db queries, which is faster, cheaper, and can be handy if we move auth to a separate service.

Reply:
But when checking permission based on the passed in shopId, you always have to remember to include `shopId` in the database find as well. Otherwise they could pass in any shop they have access to while passing in an entity ID that belongs to a different shop.